src/embind/embind.js

```diff
@@ -1347,16 +1347,33 @@ var LibraryEmbind = {
   },
 
   $init_RegisteredPointer__deps: [
+    '$RegisteredPointer',
     '$readPointer',
-    '$RegisteredPointer_getPointee', '$RegisteredPointer_destructor',
-    '$RegisteredPointer_deleteObject', '$RegisteredPointer_fromWireType'],
+    '$RegisteredPointer_fromWireType',
+    '$GenericWireTypeSize',
+  ],
   $init_RegisteredPointer: () => {
-    RegisteredPointer.prototype.getPointee = RegisteredPointer_getPointee;
-    RegisteredPointer.prototype.destructor = RegisteredPointer_destructor;
-    RegisteredPointer.prototype['argPackAdvance'] = GenericWireTypeSize;
-    RegisteredPointer.prototype['readValueFromPointer'] = readPointer;
-    RegisteredPointer.prototype['deleteObject'] = RegisteredPointer_deleteObject;
-    RegisteredPointer.prototype['fromWireType'] = RegisteredPointer_fromWireType;
+    Object.assign(RegisteredPointer.prototype, {
+      getPointee(ptr) {
+        if (this.rawGetPointee) {
+          ptr = this.rawGetPointee(ptr);
+        }
+        return ptr;
+      },
+      destructor(ptr) {
+        if (this.rawDestructor) {
+          this.rawDestructor(ptr);
+        }
+      },
+      'argPackAdvance': GenericWireTypeSize,
+      'readValueFromPointer': readPointer,
+      'deleteObject': function(handle) {
```

Comment:
FWIW if you want, and assuming Closure preserves them in the same way, you can also shorten such syntax here and in other places to

```js
'deleteObject'(handle) { ... }
````

Reply:
Done!